src/main/scala/viper/gobra/Gobra.scala

```diff
@@ -212,44 +273,49 @@ class GobraFrontend {
 
 object GobraRunner extends GobraFrontend with StrictLogging {
   def main(args: Array[String]): Unit = {
+    val executor: GobraExecutionContext = new DefaultGobraExecutionContext()
+    val verifier = createVerifier()
+    var exitCode = 0
     try {
-      val scallopGobraconfig = new ScallopGobraConfig(args.toSeq)
-      val config = scallopGobraconfig.config
-      val executor: GobraExecutionContext = new DefaultGobraExecutionContext()
-      val verifier = createVerifier()
-      val resultFuture = verifier.verify(config)(executor)
-      val result = Await.result(resultFuture, Duration.Inf)
-      executor.terminate()
+      val scallopGobraConfig = new ScallopGobraConfig(args.toSeq)
+      val config = scallopGobraConfig.config
 
-      result match {
-        case VerifierResult.Success =>
-          logger.info(s"${verifier.name} found no errors")
-          sys.exit(0)
-        case VerifierResult.Failure(errors) =>
-          logger.error(s"${verifier.name} has found ${errors.length} error(s):")
-          errors foreach (e => logger.error(s"\t${e.formattedMessage}"))
-          sys.exit(1)
+      // Set up gobra directory
+      val gobraDirectory = config.gobraDirectory.toAbsolutePath.toFile
+      if(!gobraDirectory.isDirectory) {
+        Violation.violation(gobraDirectory.mkdir(), s"Could not create directory $gobraDirectory")
+      }
+
+      // Make sure we have the correct permissions to the gobra directory
+      Violation.violation(gobraDirectory.canRead && gobraDirectory.canWrite, "Couldn't write to gobra directory " + config.gobraDirectory.toString)
```

Comment:
This error message can be misleading if there is write permission to the directory but not read permission. I would separate this in two checks.